packages/fuels-programs/src/logs.rs

```diff
@@ -9,13 +10,47 @@ use fuel_abi_types::error_codes::{
     FAILED_SEND_MESSAGE_SIGNAL, FAILED_TRANSFER_TO_ADDRESS_SIGNAL,
 };
 use fuel_tx::{ContractId, Receipt};
-use fuels_core::{traits::DecodableLog, try_from_bytes};
+use fuels_core::try_from_bytes;
 use fuels_types::{
     errors::{Error, Result},
-    param_types::ParamType,
     traits::{Parameterize, Tokenizable},
 };
 
+#[derive(Clone)]
+pub struct LogFormatter {
+    formatter: fn(&[u8]) -> Result<String>,
+    type_id: TypeId,
+}
+
+impl LogFormatter {
+    pub fn new<T: Tokenizable + Parameterize + Debug + 'static>() -> Self {
+        Self {
+            formatter: Self::format_log::<T>,
+            type_id: TypeId::of::<T>(),
+        }
+    }
+
+    fn format_log<T: Parameterize + Tokenizable + Debug>(bytes: &[u8]) -> Result<String> {
+        Ok(format!("{:?}", try_from_bytes::<T>(bytes)?))
+    }
+
+    pub fn can_handle_type<T: Tokenizable + Parameterize + 'static>(&self) -> bool {
```

Comment:
Now that we have path disambiguation, shouldnt this be called `is_type`?

Reply:
> Now that we have path disambiguation

If you're referring to the type-paths forc feature, then this isn't related. The type to decode isn't given as a `TypePath`. Rather `T` is a well-known type at the point of invocation. rustc will take care that there can be no ambiguity with regard to which type the caller meant.

> shouldnt this be called is_type


LogFormatters aren't of a type themselves, instead, they decode logs of a type.

We're asking here whether the log decoder instance in question can handle decoding logs of the given type.